src/CodeGen_Internal.cpp

```diff
@@ -474,5 +474,31 @@ void set_function_attributes_for_target(llvm::Function *fn, Target t) {
     fn->addFnAttr("reciprocal-estimates", "none");
 }
 
+Expr compile_unsafe_promises(const Call *op, bool check_unsafe_promises) {
```

Comment:
I think similar helper functions use `lower` instead of `compile`, e.g. `lower_unsafe_promises`.